src/node/historical_queries.h

```diff
@@ -508,24 +590,15 @@ namespace ccf::historical
 
           if (request.include_receipts)
           {
-            const auto result = request.populate_receipts(seqno);
-            switch (result)
+            const auto next_seqno = request.populate_receipts(seqno);
```

Comment:
This used to return an enum, because if it wanted you to fetch another entry (as a guess at a supporting signature), that was always the one after we'd currently looked at. That's no longer true!

Explanation: If we already have `{7, 8, 9}` (and maybe we already have receipts for them too), and we're now fetching `{4, 7, 8, 9}` (with receipts), and the only signature is at 11, then here's what happens:
- We ask for 4
- We get 4, validate it, but it's not a signature
- Looking for a receipt for 4, we check 7, 8, and 9 since we already have them, just in-case.
- None of those are a signature, so we fetch 5 as a supporting signature.
- We get 5, validate it, but it's not a signature.
- Still looking for a receipt for 4, our next best guess is 6
- We get 6, validate it, but it's not a signature.
- Still looking for a receipt for 4, our next best guess is _10_. In `populate_receipts`, we examined 6 (that we just got), and 7, 8, and 9 (that we already had), so returned `next_seqno = 10`.
- We get 10, validate it, but it's not a signature
- Still looking for a receipt for 4, our next best guess is 11.
- We get 11, validate it, it's a signature! Populate receipt for 4, and now we can answer the original state query.

Anyway all of this mess is only necessary because we're still linearly scanning for signatures. It should be possible to do this far more efficiently - either asking the host to return the next signature and letting it scan (potentially using ledger header bits), or indexing signatures internally,